nucliadb/tests/ingest/unit/orm/test_brain.py

```diff
@@ -202,19 +207,88 @@ def test_apply_field_paragraphs_populates_page_number():
         field_key,
         fcmw.metadata,
         page_positions=page_positions,
-        extracted_text=None,
+        extracted_text=extracted_text,
         user_field_metadata=None,
         replace_field=False,
         skip_paragraphs=False,
     )
 
     assert len(br.brain.paragraphs[field_key].paragraphs) == 2
     for paragraph in br.brain.paragraphs[field_key].paragraphs.values():
-        assert paragraph.metadata.position.page_number == 2
-        assert paragraph.metadata.position.start == 40
-        assert paragraph.metadata.position.end == 54
+        assert paragraph.metadata.position.page_number == 0
+        assert paragraph.metadata.position.start == 0
+        assert paragraph.metadata.position.end == len(text)
         assert paragraph.metadata.position.start_seconds == [0]
-        assert paragraph.metadata.position.end_seconds == [10]
+        assert paragraph.metadata.position.end_seconds == [len(text)]
+
+
+def test_apply_field_paragraphs_append_splits():
+    field_key = "t/text1"
+    split = "subfield"
+    text = "Some text here"
+
+    extracted_text = ExtractedText(split_text={split: text})
+
+    p1 = Paragraph(start=0, end=len(text), start_seconds=[0], end_seconds=[len(text)], text=text)
+    p1.sentences.append(Sentence(start=0, end=len(text)))
+
+    fcm = FieldComputedMetadata()
+    fcm.split_metadata[split].paragraphs.append(p1)
+
+    # If no append splits are passed, all splits are included in the index message
+    br = ResourceBrain(rid="foo")
+    br.apply_field_paragraphs(
+        field_key,
+        field_computed_metadata=fcm,
+        page_positions={},
+        extracted_text=extracted_text,
+        user_field_metadata=None,
+        replace_field=False,
+        skip_paragraphs=False,
+        append_splits=None,
+    )
+    assert len(br.brain.paragraphs[field_key].paragraphs) == 1
```

Comment:
Maybe it would be good to check that the brain does not contain deletions. Not sure if it can be checked easily (calling `apply_field_paragraphs` might not be enough for this to be meaningful), but we have had lots of problems with deletions so I think it might be worth being extra careful.